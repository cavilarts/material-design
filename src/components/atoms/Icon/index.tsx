import classNames from 'classnames';

import './index.scss';
import { iconProps } from './interface';

const Icon: React.FC<iconProps> = ({ name, ariaHidden, ariaLabel }) => {
  return <span className={classNames(`icon-${name}`)} aria-hidden={ariaHidden} aria-label={ariaLabel}></span>;
};

export default Icon;
