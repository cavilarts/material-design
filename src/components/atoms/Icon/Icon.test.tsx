import { shallow, configure } from 'enzyme';
import Icon from './index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('Icon', () => {
  it('should render an Icon component', () => {
    const icon = shallow(<Icon name='home' />);

    expect(icon).toMatchSnapshot();
  });

  it('should receive a name property', () => {
    const icon = shallow(<Icon name='home' />);

    expect(icon.prop('className')).toBe('icon-home');
  });

  it('should receive an ariaHidden property', () => {
    const icon = shallow(<Icon name='amazon' ariaHidden />);

    expect(icon.prop('aria-hidden')).toBe(true);
  });

  it('should receive an ariaLabel property', () => {
    const icon = shallow(<Icon name='amazon' ariaHidden={false} ariaLabel='amazon-icon' />);

    expect(icon.prop('aria-label')).toBe('amazon-icon');
  });
});
