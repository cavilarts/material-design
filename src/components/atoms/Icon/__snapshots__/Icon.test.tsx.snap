// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render an Icon component 1`] = `
<span
  className="icon-home"
/>
`;
